internal/test/e2e/tinkerbellHardwareQueue.go

```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+)
+
+// Default timeout for Tink tests to poll for hardware.
+const hwPollingTimeout = 120 * time.Minute
+
+type hardwareQueue struct {
+	hws []*api.Hardware
+	mu  sync.Mutex
+}
+
+func (hwQu *hardwareQueue) reserveHardware(count int) ([]*api.Hardware, error) {
+	now := time.Now()
+	after := now.Add(hwPollingTimeout)
+	for {
+		if now.After(after) {
+			return nil, fmt.Errorf("hardware polling request timed out")
+		}
+		hwQu.mu.Lock()
+		if count <= len(hwQu.hws) {
+			hardwareReserved := hwQu.dequeueHw(count)
+			hwQu.mu.Unlock()
+			return hardwareReserved, nil
+		}
+		hwQu.mu.Unlock()
+		time.Sleep(1 * time.Minute)
+	}
+}
+
+func (hwQu *hardwareQueue) releaseHardware(hardwareToRelease []*api.Hardware) {
```

Comment:
```suggestion
func (hwQu *hardwareQueue) releaseHardware(hw []*api.Hardware) {
```

What's the rationale for the separate `enqueueHw` and `releaseHardware` methods?

Reply:
The queue object in itself, just has two methods enqueue and dequeue which any queue is generally expected to have. Since its a concurrent queue, it also has an inherent lock. The locking and unlocking to make the write or enqueuing (thread safe) is handled by `realeaseHardware` . This is just for modularity and separating out the queue functionality from our reservation logic.